### PR TITLE
IGNITE-23077 Fix flaky ItIndexNodeFinishedRwTransactionsCheckerTest#testOnePhaseCommitViaKeyValue

### PR DESCRIPTION
**Pull request by tkalkirill**
https://issues.apache.org/jira/browse/IGNITE-23077

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/index/src/integrationTest/java/org/apache/ignite/internal/index/ItIndexNodeFinishedRwTransactionsCheckerTest.java [hunk 1]
```diff
@@ -64,7 +64,7 @@ public class ItIndexNodeFinishedRwTransactionsCheckerTest extends ClusterPerClas
 
     private static final String TABLE_NAME = "TEST_TABLE";
 
-    private static String zoneNameForUpdateCatalogVersionOnly = "FAKE_TEST_ZONE";
+    private String zoneNameForUpdateCatalogVersionOnly = "FAKE_TEST_ZONE";
```
Comment (Phillippko) [inline modules/index/src/integrationTest/java/org/apache/ignite/internal/index/ItIndexNodeFinishedRwTransactionsCheckerTest.java:67]: Let's make it a constant and use `dropAllTables `+ `dropAllZonesExceptDefaultOne` in tearDown method

File: modules/index/src/integrationTest/java/org/apache/ignite/internal/index/ItIndexNodeFinishedRwTransactionsCheckerTest.java [hunk 2]
```diff
@@ -251,7 +251,7 @@ private static long differences(long[] partitionSizes0, long[] partitionsSizes1)
                 .count();
     }
 
-    private static void fakeUpdateCatalog() {
+    private void fakeUpdateCatalog() {
```
Comment (Phillippko) [inline modules/index/src/integrationTest/java/org/apache/ignite/internal/index/ItIndexNodeFinishedRwTransactionsCheckerTest.java:254]: Let's revert it (see previous comment)

File: modules/index/src/integrationTest/java/org/apache/ignite/internal/index/ItIndexNodeFinishedRwTransactionsCheckerTest.java [hunk 3]
```diff
@@ -74,8 +74,8 @@ protected int initialNodes() {
     @BeforeEach
     void setUp() {
         if (node() != null) {
```
Comment (Phillippko) [inline modules/index/src/integrationTest/java/org/apache/ignite/internal/index/ItIndexNodeFinishedRwTransactionsCheckerTest.java:76]: Why we need this check? If node is null, is it a valid state to start a test? 

I removed checks from tearDown / beforeEach and works like a charm

**Review by Phillippko — APPROVED** [review]
Minor comments

File: modules/index/src/integrationTest/java/org/apache/ignite/internal/index/ItIndexNodeFinishedRwTransactionsCheckerTest.java (hunk 1 above)
Comment (tkalkirill) [reply modules/index/src/integrationTest/java/org/apache/ignite/internal/index/ItIndexNodeFinishedRwTransactionsCheckerTest.java:67]: Why not.

File: modules/index/src/integrationTest/java/org/apache/ignite/internal/index/ItIndexNodeFinishedRwTransactionsCheckerTest.java (hunk 2 above)
Comment (tkalkirill) [reply modules/index/src/integrationTest/java/org/apache/ignite/internal/index/ItIndexNodeFinishedRwTransactionsCheckerTest.java:254]: Why not.

File: modules/index/src/integrationTest/java/org/apache/ignite/internal/index/ItIndexNodeFinishedRwTransactionsCheckerTest.java (hunk 3 above)
Comment (tkalkirill) [reply modules/index/src/integrationTest/java/org/apache/ignite/internal/index/ItIndexNodeFinishedRwTransactionsCheckerTest.java:76]: I also don’t see the point in checking, I removed it, thanks!

**Pull request merged**